system-tests/lib/cre/features/ocr/ocr.go

```diff
@@ -0,0 +1,528 @@
+	if err != nil {
+		return errors.Wrap(err, "failed to get OCR3 capability address")
+	}
+
+	donTimeKey := datastore.NewAddressRefKey(
+		creEnv.DonTopology.HomeChainSelector,
+		datastore.ContractType(keystone_changeset.OCR3Capability.String()),
+		semver.MustParse("1.0.0"),
+		contracts.DONTimeContractQualifier,
+	)
+	donTimeAddress, err := creEnv.CldfEnvironment.DataStore.Addresses().Get(donTimeKey)
+	if err != nil {
+		return errors.Wrap(err, "failed to get DON Time address")
+	}
+
+	jobSpecs := []*job.ProposeJobRequest{}
+
+	chainID, chErr := chainselectors.ChainIdFromSelector(creEnv.DonTopology.HomeChainSelector)
+	if chErr != nil {
+		return errors.Wrapf(chErr, "failed to get chain ID from chain selector %d", creEnv.DonTopology.HomeChainSelector)
+	}
+
+	for _, don := range creEnv.DonTopology.Dons.List() {
+		if !don.HasFlag(flag) {
+			continue
+		}
+
+		workerNodes, wErr := don.Workers()
+		if wErr != nil {
+			return errors.Wrap(wErr, "failed to find worker nodes")
+		}
+
+		bootstrapNode, isBootstrap := creEnv.DonTopology.BootstrapNode()
+		if !isBootstrap {
+			return errors.New("could not find bootstrap node in topology, exactly one bootstrap node is required")
+		}
+
+		_, ocrPeeringCfg, err := cre.PeeringCfgs(bootstrapNode)
+		if err != nil {
+			return errors.Wrap(err, "failed to get peering configs")
+		}
+
+		jobSpecs = append(jobSpecs, jobs.BootstrapOCR3(bootstrapNode.JobDistributorDetails.NodeID, "ocr3-capability", ocr3CapabilityAddress.Address, chainID))
+
+		for _, workerNode := range workerNodes {
+			evmKey, ok := workerNode.Keys.EVM[chainID]
+			if !ok {
+				return fmt.Errorf("failed to get EVM key (chainID %d, node index %d)", chainID, workerNode.Index)
+			}
+
+			// we need the OCR2 key bundle for the EVM chain, because OCR jobs currently run only on EVM chains
+			evmOCR2KeyBundle, ok := workerNode.Keys.OCR2BundleIDs[chainselectors.FamilyEVM]
+			if !ok {
+				return fmt.Errorf("node %s does not have OCR2 key bundle for evm", workerNode.Name)
+			}
+
+			// we pass here bundles for all chains to enable multi-chain signing
+			jobSpecs = append(jobSpecs, jobs.WorkerOCR3(workerNode.JobDistributorDetails.NodeID, ocr3CapabilityAddress.Address, evmKey.PublicAddress.Hex(), evmOCR2KeyBundle, workerNode.Keys.OCR2BundleIDs, ocrPeeringCfg, chainID))
+			jobSpecs = append(jobSpecs, jobs.DonTimeJob(workerNode.JobDistributorDetails.NodeID, donTimeAddress.Address, evmKey.PublicAddress.Hex(), evmOCR2KeyBundle, ocrPeeringCfg, chainID))
+		}
+	}
+
+	return jobs.Create(ctx, creEnv.CldfEnvironment.Offchain, creEnv.DonTopology, jobSpecs)
+}
+
+// TODO make our DON use clclient from the CTF instead, since then we could remove nodeSetOutput as DON has Nodes that would have this client initialised
+func waitForLogPollerToBeHealthy(dons []*cre.DON, nodeSetOutput []*cre.WrappedNodeOutput) error {
+	for idx, nodeSetOut := range nodeSetOutput {
+		if !dons[idx].HasFlag(cre.ConsensusCapability) || !dons[idx].HasFlag(cre.VaultCapability) {
```

Comment:
:( Not the right place to be checking this -- can we make it easier to know exactly what nodes we are dealing with?

Reply:
This was copy pasta, in reality we should probably return early if no DON has that capability.